src/pip/_internal/utils/appdirs.py

```diff
@@ -22,255 +20,22 @@
-                for x in xdg_config_dirs.split(os.pathsep)
-            ]
-        else:
-            pathlist = []
-
-        # always look in /etc directly as well
-        pathlist.append('/etc')
-
-    return pathlist
-
-
-# -- Windows support functions --
-
-def _get_win_folder_from_registry(csidl_name):
-    # type: (str) -> str
-    """
-    This is a fallback technique at best. I'm not sure if using the
-    registry for this guarantees us the correct answer for all CSIDL_*
-    names.
-    """
-    import _winreg
-
-    shell_folder_name = {
-        "CSIDL_APPDATA": "AppData",
-        "CSIDL_COMMON_APPDATA": "Common AppData",
-        "CSIDL_LOCAL_APPDATA": "Local AppData",
-    }[csidl_name]
-
-    key = _winreg.OpenKey(
-        _winreg.HKEY_CURRENT_USER,
-        r"Software\Microsoft\Windows\CurrentVersion\Explorer\Shell Folders"
-    )
-    directory, _type = _winreg.QueryValueEx(key, shell_folder_name)
-    return directory
-
-
-def _get_win_folder_with_ctypes(csidl_name):
-    # type: (str) -> str
-    # On Python 2, ctypes.create_unicode_buffer().value returns "unicode",
-    # which isn't the same as str in the annotation above.
-    csidl_const = {
-        "CSIDL_APPDATA": 26,
-        "CSIDL_COMMON_APPDATA": 35,
-        "CSIDL_LOCAL_APPDATA": 28,
-    }[csidl_name]
-
-    buf = ctypes.create_unicode_buffer(1024)
-    windll = ctypes.windll  # type: ignore
-    windll.shell32.SHGetFolderPathW(None, csidl_const, None, 0, buf)
-
-    # Downgrade to short path name if have highbit chars. See
-    # <http://bugs.activestate.com/show_bug.cgi?id=85099>.
-    has_high_char = False
-    for c in buf:
-        if ord(c) > 255:
-            has_high_char = True
-            break
-    if has_high_char:
-        buf2 = ctypes.create_unicode_buffer(1024)
-        if windll.kernel32.GetShortPathNameW(buf.value, buf2, 1024):
-            buf = buf2
-
-    # The type: ignore is explained under the type annotation for this function
-    return buf.value  # type: ignore
-
-
-if WINDOWS:
-    try:
-        import ctypes
-        _get_win_folder = _get_win_folder_with_ctypes
-    except ImportError:
-        _get_win_folder = _get_win_folder_from_registry
-
-
-def _win_path_to_bytes(path):
-    """Encode Windows paths to bytes. Only used on Python 2.
-
-    Motivation is to be consistent with other operating systems where paths
-    are also returned as bytes. This avoids problems mixing bytes and Unicode
-    elsewhere in the codebase. For more details and discussion see
-    <https://github.com/pypa/pip/issues/3463>.
-
-    If encoding using ASCII and MBCS fails, return the original Unicode path.
-    """
-    for encoding in ('ASCII', 'MBCS'):
-        try:
-            return path.encode(encoding)
-        except (UnicodeEncodeError, LookupError):
-            pass
-    return path
+    dirval = _appdirs.site_config_dir(appname, appauthor=False, multipath=True)
+    if _appdirs.system == "linux2":
+        return dirval.split(os.pathsep)
```

Comment:
This will return `['pip', '/etc']` when `XDG_CONFIG_DIRS` is empty, whereas the previous implementation would return just `['/etc']`.

Reply:
Added a patch to match the previous behaviour. Or should I treat this at the call site instead? Something like (looks quite hacky):

```python
def site_config_dirs(appname):
    # type: (str) -> List[str]
    dirval = _appdirs.site_config_dir(appname, appauthor=False, multipath=True)
    if _appdirs.system not in ["win32", "darwin"]:
        return [x for x in dirval.split(os.pathsep) if x != appname]
    return [dirval]
```

Reply:
From [here](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html) it looks like both appdirs implementations were wrong. :sweat_smile:

>  If `$XDG_CONFIG_DIRS` is either not set or empty, a value equal to `/etc/xdg` should be used.

In this case the right place to put it is definitely the vendored appdirs, like

```
         path = os.getenv('XDG_CONFIG_DIRS', '/etc/xdg')
         if not path:
             path = '/etc/xdg'
```

Reply:
Thanks for digging up that spec! I was wondering if an empty value should be treated as unset (since it’s how most tools do), but thought appdirs should have a reason. Turns out it probably doesn’t :p

I’ll probably do with `os.getenv('XDG_CONFIG_DIRS') or '/etc/xdg'`.

Now I’m wondering whether we need to patch other XDG usages in appdirs as well. They are all wrong according to the spec…

Reply:
IMO just this one is OK for this PR - at least then we're at parity with what we had in pip (+this fix).